### PR TITLE
[DTSSTCI-95] Add hearingCreatedDate to a listing

### PR DESCRIPTION
**Pull request by oleon23**
https://tools.hmcts.net/jira/browse/DTSSTCI-95 

- [ ] tests have been updated / new tests has been added (if needed)

Updated listings:
caseListed
hearingCancelled
hearingCompleted
hearingPostponed

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/model/Listing.java [hunk 1]
```diff
@@ -38,6 +39,12 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class Listing {
 
+    @CCD(
+        label = "Hearing Created",
+        access = {DefaultAccess.class, CaseworkerWithCAAAccess.class}
+    )
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/model/Listing.java:45]: Do we need a particular format?
`@JsonFormat(pattern = "yyyy-MM-dd")`

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/helper/RecordListHelper.java [hunk 2]
```diff
@@ -126,25 +130,28 @@ public Listing checkAndUpdateVenueInformation(Listing listing) {
...
     public Listing saveSummary(CaseData caseData) {
-        caseData.getListing().setHearingFormat(caseData.getListing().getHearingFormat());
-        caseData.getListing().setHearingType(caseData.getListing().getHearingType());
-        caseData.getListing().getSummary().setSubjectName(caseData.getCicCase().getFullName());
-        caseData.setCurrentEvent("");
-        if (null != caseData.getListing()
-            && null != caseData.getListing().getNumberOfDays()
-            && caseData.getListing().getNumberOfDays().equals(YesOrNo.NO)) {
-            caseData.getListing().setAdditionalHearingDate(null);
+        final Listing listing = caseData.getListing();
```
Comment (oleon23) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/helper/RecordListHelper.java:142]: Refactored slightly for better readability and testing. Though further improvement could be to refactor all `checkAndUpdateVenueInformation`, `checkAndUpdateVenueInformationSummary` and `SaveSummary` to all be void methods as they only  modify the `caseData.getListing()` object.

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/model/Listing.java [hunk 3]
```diff
@@ -38,12 +38,62 @@
...
+    @JsonFormat(pattern = "yyyy-MM-dd")
+    private LocalDate hearingCreatedDate;
+
     @CCD(
         label = "Hearing Status",
         access = {DefaultAccess.class, CaseworkerWithCAAAccess.class}
     )
     private HearingState hearingStatus;
 
+    @CCD(
```
Comment (oleon23) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/model/Listing.java:54]: This change (L54:96) was mainly to reorder the fields in the Listing tab page.

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

**Review by tomxelliott — DISMISSED** [review]

**Review by tomxelliott — APPROVED** [review]

**Review by gmmagruder — APPROVED** [review]

**Comment by JakeCG** [comment]
QA Approved.

**Pull request merged**